File: searchInsertPosition.py
# problem: https://leetcode.com/problems/search-insert-position/
# Runtime: 68 ms, faster than 18.74% of Python3 online submissions for Search Insert Position.
# Memory Usage: 15 MB, less than 56.94% of Python3 online submissions for Search Insert Position.

from typing import List


class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums)

        while l < r:

            mid = ((r - l) // 2) + l
            if nums[mid] < target:
                mid += 1
                l = mid
            else:
                r = mid

        return mid

File: searchInsertPosition.cpp
// problem: https://leetcode.com/problems/search-insert-position/description/
// 0 ms
// Beats
// 100%
// Memory
// 9.6 MB
// Beats
// 96.30%

#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
  int searchInsert(vector<int> &nums, int target) {
    return lower_bound(nums.begin(), nums.end(), target) - nums.begin();
  }
};
